includes/Export.php

```diff
@@ -559,83 +731,103 @@ function writeRevision( $row ) {
+		) {
+			$out .= "      " . Xml::element( 'sha1', null, strval( $row->rev_sha1 ) ) . "\n";
+		} else {
+			$out .= "      <sha1/>\n";
+		}
+
 		Hooks::run( 'XmlDumpWriterWriteRevision', array( &$this, &$out, $row, $text ) );
 
 		$out .= "    </revision>\n";
 
-		wfProfileOut( __METHOD__ );
 		return $out;
 	}
 
 	/**
-	 * Dumps a <logitem> section on the output stream, with
+	 * Dumps a "<logitem>" section on the output stream, with
 	 * data filled in from the given database row.
 	 *
-	 * @param $row object
+	 * @param object $row
 	 * @return string
 	 * @access private
 	 */
 	function writeLogItem( $row ) {
-		wfProfileIn( __METHOD__ );
 
-		$out  = "    <logitem>\n";
-		$out .= "      " . Xml::element( 'id', null, strval( $row->log_id ) ) . "\n";
+		$out = "  <logitem>\n";
+		$out .= "    " . Xml::element( 'id', null, strval( $row->log_id ) ) . "\n";
 
-		$out .= $this->writeTimestamp( $row->log_timestamp );
+		$out .= $this->writeTimestamp( $row->log_timestamp, "    " );
 
 		if ( $row->log_deleted & LogPage::DELETED_USER ) {
-			$out .= "      " . Xml::element( 'contributor', array( 'deleted' => 'deleted' ) ) . "\n";
+			$out .= "    " . Xml::element( 'contributor', array( 'deleted' => 'deleted' ) ) . "\n";
 		} else {
-			$out .= $this->writeContributor( $row->log_user, $row->user_name );
+			$out .= $this->writeContributor( $row->log_user, $row->user_name, "    " );
 		}
 
 		if ( $row->log_deleted & LogPage::DELETED_COMMENT ) {
-			$out .= "      " . Xml::element( 'comment', array( 'deleted' => 'deleted' ) ) . "\n";
+			$out .= "    " . Xml::element( 'comment', array( 'deleted' => 'deleted' ) ) . "\n";
 		} elseif ( $row->log_comment != '' ) {
-			$out .= "      " . Xml::elementClean( 'comment', null, strval( $row->log_comment ) ) . "\n";
+			$out .= "    " . Xml::elementClean( 'comment', null, strval( $row->log_comment ) ) . "\n";
 		}
 
-		$out .= "      " . Xml::element( 'type', null, strval( $row->log_type ) ) . "\n";
-		$out .= "      " . Xml::element( 'action', null, strval( $row->log_action ) ) . "\n";
+		$out .= "    " . Xml::element( 'type', null, strval( $row->log_type ) ) . "\n";
+		$out .= "    " . Xml::element( 'action', null, strval( $row->log_action ) ) . "\n";
 
 		if ( $row->log_deleted & LogPage::DELETED_ACTION ) {
-			$out .= "      " . Xml::element( 'text', array( 'deleted' => 'deleted' ) ) . "\n";
+			$out .= "    " . Xml::element( 'text', array( 'deleted' => 'deleted' ) ) . "\n";
 		} else {
 			$title = Title::makeTitle( $row->log_namespace, $row->log_title );
-			$out .= "      " . Xml::elementClean( 'logtitle', null, self::canonicalTitle( $title ) ) . "\n";
-			$out .= "      " . Xml::elementClean( 'params',
+			$out .= "    " . Xml::elementClean( 'logtitle', null, self::canonicalTitle( $title ) ) . "\n";
+			$out .= "    " . Xml::elementClean( 'params',
 				array( 'xml:space' => 'preserve' ),
 				strval( $row->log_params ) ) . "\n";
 		}
 
-		$out .= "    </logitem>\n";
+		$out .= "  </logitem>\n";
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Immediately return this expression instead of assigning it to the temporary variable "$out". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1488)